code/modules/surgery/helpers.dm

```diff
@@ -0,0 +1,95 @@
+/proc/attempt_initiate_surgery(obj/item/I, mob/living/M, mob/user, var/override )
+	if(istype(M))
+		var/mob/living/carbon/human/H
+		var/obj/item/organ/external/affecting
+		var/selected_zone = user.zone_sel.selecting
+		if(istype(M, /mob/living/carbon/human))
+			H = M
+			affecting = H.get_organ(check_zone(selected_zone))
+
+		if(can_operate(H))	//if they're prone or a slime
```

Comment:
I found the problem with your xeno surgery - with xenos, H is never set, and so `null` is passed to can_operate, which causes a runtime.